# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
air_bnb = pd.read_csv('AB_NYC_2019.csv')
air_bnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [7]:
# How many neighborhood groups are available and which shows up the most?
hoods = air_bnb['neighbourhood_group']
hoods.head()
total = max(range(len(hoods)))
print(total)
# Fun to find the neighborhood with most listings (and # of listings)
def mostHoods(df):
    hood_stats = {}
    for hood in hoods:
        if hood in hood_stats.keys():
            hood_stats[hood] += 1
        else:
            hood_stats[hood] = 1
    for k,v in hood_stats.items():
        if v == max(hood_stats.values()):
            return k, v
print(mostHoods(hoods))
# Fun to count the num of diff neighborhoods
def numHoods(df):
    hood_stats = {}
    for hood in hoods:
        if hood in hood_stats.keys():
            hood_stats[hood] += 1
        else:
            hood_stats[hood] = 1
    nh = ''
    count = 0
    for hood in hood_stats.keys():
        if hood == nh:
            continue
        else:
            count += 1
            nh = hood
    return count
print(numHoods(hoods))


48894
('Manhattan', 21661)
5


In [13]:
# Are private rooms the most popular in manhattan?
rooms = air_bnb[['neighbourhood_group', 'room_type']]

private = rooms[rooms['room_type'].str.contains("Private")]
privateHoods = private['neighbourhood_group']

def mostPrivateRooms(df):
    hood_stats = {}
    for hood in df:
        if hood in hood_stats.keys():
            hood_stats[hood] += 1
        else:
            hood_stats[hood] = 1
    for k,v in hood_stats.items():
        if v == max(hood_stats.values()):
            return f'Private rooms are most popular in {k}: {v}'
print(mostPrivateRooms(privateHoods))

Private rooms are most popular in Brooklyn: 10132


In [15]:
# Which hosts are the busiest and based on their reviews?

host_reviews = air_bnb[['host_name', 'number_of_reviews']]
#print(host_reviews.head(10))
host_reviews.sort_values('number_of_reviews')

host_name  number_of_reviews
48894  Christophe                  0
28711     Eleanor                  0
15661      Joseph                  0
40011          辣辣                  0
7707          Aya                  0
...           ...                ...
13495        Dona                576
2015           Jj                594
2030           Jj                597
2031           Jj                607
11759        Dona                629

[48895 rows x 2 columns]

In [33]:
#Which neighorhood group has the highest average price?
#print(air_bnb.keys())
#hoods = air_bnb['neighbourhood_group']
#print(hoods.head())
#get all hoods
def getHoods(df):
    stats = {}
    for hood in df:
        if hood in stats.keys():
            stats[hood] += 1
        else:
            stats[hood] = 1
    return stats
#print(getHoods(hoods))

#print(air_bnb[['neighbourhood_group', 'price']])
air_bnb.groupby('neighbourhood_group').mean()[['price']]

price
neighbourhood_group            
Bronx                 87.496792
Brooklyn             124.383207
Manhattan            196.875814
Queens                99.517649
Staten Island        114.812332

In [39]:
# Which neighbor hood group has the highest total price?
air_bnb[['neighbourhood_group', 'price']].sort_values('price')

neighbourhood_group  price
25796            Brooklyn      0
25634            Brooklyn      0
25433               Bronx      0
25753            Brooklyn      0
23161            Brooklyn      0
...                   ...    ...
40433           Manhattan   9999
12342           Manhattan   9999
17692            Brooklyn  10000
9151               Queens  10000
29238           Manhattan  10000

[48895 rows x 2 columns]

In [40]:
#Which top 5 hosts have the highest total price?
air_bnb[['host_name','price']].sort_values('price')

host_name  price
25796       Adeyemi      0
25634  Martial Loft      0
25433        Anisha      0
25753        Lauren      0
23161      Kimberly      0
...             ...    ...
40433          Matt   9999
12342           Amy   9999
17692          Erin  10000
9151       Kathrine  10000
29238        Jelena  10000

[48895 rows x 2 columns]

In [49]:
# Who currently has no (zero) availability with a review count of 100 or more?
# air_bnb.keys()
# avail_hosts = air_bnb[['host_name','availability_365','number_of_reviews']]

zero_avail_hosts = air_bnb[air_bnb['availability_365']==0]
zero_avail_hosts[zero_avail_hosts['number_of_reviews']>=100]


id                                       name    host_id  \
8          5203         Cozy Clean Guest Room - Family Apt       7490   
94        20913         Charming 1 bed GR8 WBurg LOCATION!      79402   
132       30031           NYC artists’ loft with roof deck     129352   
174       44221             Financial District Luxury Loft     193722   
180       45556      Fort Greene, Brooklyn: Center Bedroom      67778   
...         ...                                        ...        ...   
29581  22705516         The Quietest Block in Manhattan :)  127740507   
30461  23574142                          queens get away!!  176185168   
31250  24267706  entire sunshine of the spotless mind room   21074914   
32670  25719044                COZY Room for Female Guests   40119874   
35014  27759146     Cozy corner near Empire State Building  209549523   

        host_name neighbourhood_group              neighbourhood  latitude  \
8       MaryEllen           Manhattan            Upper West Side  40.80178   
94     Christiana            Brooklyn               Williamsburg  40.70984   
132           Sol            Brooklyn                 Greenpoint  40.73494   
174         Coral           Manhattan         Financial District  40.70666   
180          Doug            Brooklyn                Fort Greene  40.68863   
...           ...                 ...                        ...       ...   
29581    Kathleen           Manhattan                     Harlem  40.83102   
30461       Janet              Queens                  Laurelton  40.68209   
31250      Albert            Brooklyn         Bedford-Stuyvesant  40.68234   
32670    Stephany            Brooklyn  Prospect-Lefferts Gardens  40.66242   
35014     Mariluz           Manhattan                    Midtown  40.74858   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
8      -73.96723     Private room     79               2                118   
94     -73.95775  Entire home/apt    100               5                168   
132    -73.95030     Private room     50               3                193   
174    -74.01374  Entire home/apt    196               3                114   
180    -73.97691     Private room     65               2                206   
...          ...              ...    ...             ...                ...   
29581  -73.94181     Private room     65               2                103   
30461  -73.73662     Private room     65               1                119   
31250  -73.91318     Private room     49               1                102   
32670  -73.94417     Private room     48               1                131   
35014  -73.98341      Shared room     62               1                112   

      last_review  reviews_per_month  calculated_host_listings_count  \
8      2017-07-21               0.99                               1   
94     2018-07-22               1.57                               1   
132    2019-05-20               1.86                               1   
174    2019-06-20               1.06                               1   
180    2019-06-30               1.92                               2   
...           ...                ...                             ...   
29581  2019-07-07               5.89                               2   
30461  2018-12-24               7.79                               1   
31250  2019-07-05               6.73                               3   
32670  2019-05-31               9.97                               2   
35014  2019-06-13              10.77                               3   

       availability_365  
8                     0  
94                    0  
132                   0  
174                   0  
180                   0  
...                 ...  
29581                 0  
30461                 0  
31250                 0  
32670                 0  
35014                 0  

[162 rows x 16 columns]

In [41]:
# What host has the highest total of prices and where are they located?

air_bnb[['host_name','price','neighbourhood_group']].sort_values('price')
#Jelena in Manhattan

host_name  price neighbourhood_group
25796       Adeyemi      0            Brooklyn
25634  Martial Loft      0            Brooklyn
25433        Anisha      0               Bronx
25753        Lauren      0            Brooklyn
23161      Kimberly      0            Brooklyn
...             ...    ...                 ...
40433          Matt   9999           Manhattan
12342           Amy   9999           Manhattan
17692          Erin  10000            Brooklyn
9151       Kathrine  10000              Queens
29238        Jelena  10000           Manhattan

[48895 rows x 3 columns]

In [56]:
# When did Danielle from Queens last receive a review?

print(air_bnb.keys())
air_bnb_queens = air_bnb[air_bnb['neighbourhood_group']=='Queens']
danielle = air_bnb_queens[air_bnb_queens['host_name']=='Danielle']
danielle['last_review'].sort_values()
#2019-07-08

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')


27021    2018-01-02
33861    2019-06-20
7086     2019-07-03
20403    2019-07-06
22068    2019-07-06
21517    2019-07-07
22469    2019-07-08
16349           NaN
Name: last_review, dtype: object

## Further Questions

1. Which host has the most listings?

In [57]:
air_bnb[['host_name','calculated_host_listings_count']].sort_values('calculated_host_listings_count')

host_name  calculated_host_listings_count
24447          Juan                               1
27993          Lisa                               1
27995          Tina                               1
27996     Alejandro                               1
27999       Claudia                               1
...             ...                             ...
41466  Sonder (NYC)                             327
41467  Sonder (NYC)                             327
41468  Sonder (NYC)                             327
41470  Sonder (NYC)                             327
41445  Sonder (NYC)                             327

[48895 rows x 2 columns]

2. How many listings have completely open availability?

In [60]:
air_bnb[air_bnb['availability_365']==365].count()
#1295

id                                1295
name                              1294
host_id                           1295
host_name                         1295
neighbourhood_group               1295
neighbourhood                     1295
latitude                          1295
longitude                         1295
room_type                         1295
price                             1295
minimum_nights                    1295
number_of_reviews                 1295
last_review                        841
reviews_per_month                  841
calculated_host_listings_count    1295
availability_365                  1295
dtype: int64

3. What room_types have the highest review numbers?

In [63]:
air_bnb[['number_of_reviews','room_type']].sort_values('number_of_reviews')
air_bnb.groupby(['room_type']).mean().sort_values('number_of_reviews')
#private room at 24 (avg)

id       host_id   latitude  longitude       price  \
room_type                                                                       
Shared room      2.300378e+07  1.026241e+08  40.730514 -73.943343   70.127586   
Entire home/apt  1.843818e+07  6.175593e+07  40.728649 -73.960696  211.794246   
Private room     1.946893e+07  7.247514e+07  40.729208 -73.942924   89.780973   

                 minimum_nights  number_of_reviews  reviews_per_month  \
room_type                                                               
Shared room            6.475000          16.600000           1.471726   
Entire home/apt        8.506907          22.842418           1.306578   
Private room           5.377900          24.112962           1.445209   

                 calculated_host_listings_count  availability_365  
room_type                                                          
Shared room                            4.662931        162.000862  
Entire home/apt                       10.698335        111.920304  
Private room                           3.227717        111.203933

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

1. How many neighborhood groups are available and which shows up the most?
a) 5, b) Manhattan with 21661 listings

2. Are private rooms the most popular in Manhattan?
No, private rooms are most popular in Brooklyn, with 10132 private rooms listings

3. Which hosts are the busiest and based on their reviews?
Dona and Jj

4. Which neighorhood group has the highest average price?
Manhattan at 197

5. Which neighbor hood group has the highest total price?
Manhattan at 10000

6. Which top 5 hosts have the highest total price?
Matt, Amy, Erin, Kathrine, Jelena

7. Who currently has no (zero) availability with a review count of 100 or more?
There 162 such hosts. See cell output for details.

8. What host has the highest total of prices and where are they located?
Jelena, Manhattan

9. When did Danielle from Queens last receive a review?
2019-07-08

10. Which host has the most listings?
Sonder (NYC)

11. How many listings have completely open availability?
1295

12. What room_types have the highest review numbers?
Private room (average of 24 reviews)


Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --